libraries/AP_GPS/AP_GPS.h

```diff
@@ -129,7 +130,7 @@ class AP_GPS
      */
     struct GPS_State {
         uint8_t instance; // the instance number of this GPS
-
+        uint8_t devid;  // device id of the instance
```

Comment:
please use the existing AP_HAL/Device.h method for creating dev IDs, adding BUS_TYPE_UART, and fill in the devid in a way consistent with AP_InertialSensor and AP_Compass, with make_bus_id().